FinSearchUnified/Subscriber/Widgets.php

```diff
@@ -33,4 +93,61 @@ public function beforeListingCountAction(Enlight_Hook_HookArgs $args)
             }
         }
     }
+
+    /**
+     * @param Enlight_Controller_Request_RequestHttp $request
+     *
+     * @return string
+     * @throws Exception
+     */
+    private function parseReferer($request)
+    {
+        $referrer = $request->headers->get('referer');
+        $url = parse_url($referrer, PHP_URL_PATH);
+
+        $basePath = $request->getBasePath();
+        if ($basePath !== '') {
+            $basePath = rtrim($basePath, '/') . '/';
+            $url = str_replace($basePath, '', $url);
+        }
+        $url = ltrim($url, '/');
+
+        return $url;
+    }
+
+    /**
+     * Method to match the given URL if it is a category page and set the isCategoryPage session variable accordingly
```

Comment:
Try to avoid things like `Method` in a docblock as you do not want to know what something is, but rather what it does.